### PR TITLE
Fix Keyword predicate to check nested FObject

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-1688

## Changes
- Handle keyword searching only on the the first level of object's nested FObjectProperty

### CODE REVIEW

File: src/foam/mlang/mlang.js [hunk 1]
```diff
@@ -2323,11 +2323,49 @@ foam.CLASS({
     }
   ],
 
+  properties: [
+    {
+      class: 'Boolean',
+      name: '_checkingNestedFObject',
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js]: Can you rename this checkingNextedFObject_?

File: src/foam/mlang/mlang.js [hunk 2]
```diff
@@ -2323,11 +2323,49 @@ foam.CLASS({
...
+        arg = arg.toLowerCase();
+
+        try {
+          var s = '';
+          const props = obj.cls_.getAxiomsByClass(foam.core.Property);
+          for ( let i = 0; i < props.length; i++ ) {
+            const prop = props[i];
+            if ( this.FObjectProperty.isInstance(prop) )
+              if ( this.checkNestedFObject(prop.f(obj)) ) return true;
+            else if ( this.Enum.isInstance(prop) )
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js]: Is this else and else for the line above or the line above that? Add braces to clarify.

File: src/foam/mlang/mlang.js [hunk 3]
```diff
@@ -2323,11 +2323,49 @@ foam.CLASS({
...
+              s = prop.f(obj).label.toLowerCase();
+            else if ( this.Long.isInstance(prop) )
+              s = prop.f(obj).toString().toLowerCase();
+            else if ( this.Date.isInstance(prop) )
+              s = prop.f(obj).toISOString().toLowerCase();
+            else if ( ! this.String.isInstance(prop) )
+              continue
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js]: Missing ;

File: src/foam/mlang/mlang.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/mlang/mlang.js]: Fixed in https://github.com/foam-framework/foam2/pull/3796/commits/15fbc8b03b41b76ca19e4ed4e9e41fa3a0ccaaf3.

File: src/foam/mlang/mlang.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/mlang/mlang.js]: Fixed in https://github.com/foam-framework/foam2/pull/3796/commits/ab66fa4be1959bd719e050f38aa0c4fc8cc9d43e.

File: src/foam/mlang/mlang.js (hunk 3 above)
Comment (lchanmann) [reply src/foam/mlang/mlang.js]: Fixed in https://github.com/foam-framework/foam2/pull/3796/commits/ab66fa4be1959bd719e050f38aa0c4fc8cc9d43e.

**Pull request merged**